src/frontend/src/utils/stringManipulation.ts

```
@@ -72,6 +72,52 @@ function validCommands(str: string): string {
+function sanitizeMcpName(str: string, maxLength: number = 46): string {
+  if (!str || !str.trim()) {
+    return "";
+  }
+
+  let name = str;
+
+  // Remove emojis using standard regex patterns (without unicode flags)
+  // This covers most common emoji ranges using surrogate pairs
+  name = name.replace(/[\uD83C-\uDBFF][\uDC00-\uDFFF]/g, ""); // Most emojis
+  name = name.replace(/[\u2600-\u27BF]/g, ""); // Misc symbols and dingbats
+  name = name.replace(/[\uFE00-\uFE0F]/g, ""); // Variation selectors
```

Biome (1.9.4)

[error] A character class cannot match a character and a combining character.

A character and a combining character forms a new character. Replace the character class with an alternation.

(lint/suspicious/noMisleadingCharacterClass)